src/main/java/internet/store/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,195 @@
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("User " + userId + " not updated", ex);
+        }
+    }
+
+    private User getUserByLogin(String login) {
+        String query = "SELECT user_id, user_name, login, password FROM users "
+                + "WHERE login = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            return retrieveDataFromDb(resultSet);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("User with login " + login
+                    + "is not found", ex);
+        }
+    }
+
+    private User getUserById(Long id) {
+        String query = "SELECT user_id, user_name, login, password FROM users "
+                + " WHERE user_id = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            User user = retrieveDataFromDb(resultSet);
+            return user;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("User with id " + id
+                    + " is not found", ex);
+        }
+    }
+
+    private User retrieveDataFromDb(ResultSet resultSet) throws SQLException {
+        long userId = resultSet.getLong("user_id");
+        User user = new User(userId, resultSet.getString("user_name"),
+                resultSet.getString("login"), resultSet.getString("password"));
+        return user;
+    }
+
+    private Set<Role> getUserRoles(Long userId) {
+        String query = "SELECT role_name, role_id FROM roles "
+                + " JOIN users_roles USING (role_id) WHERE user_id = ?;";
+        Set<Role> roles = new HashSet<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                roles.add(new Role(resultSet.getLong("role_id"),
+                        Role.RoleName.valueOf(resultSet.getString("role_name"))));
+            }
+            return roles;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("no users in database", ex);
+        }
+    }
+
+    private boolean defineRoles(Long userId, String role) {
```

Comment:
```suggestion
    private boolean defineRoles(User user) {
```
you can get that info from model, no need to pass to parameters